### PR TITLE
Fix random/target mods not working in spectator

### LINKED ISSUE #13751: Multiplayer spectator view broken when someone uses a mod with a random seed
When spectating in a multiplayer lobby (have not tested with normal spectating) and someone uses the "random" mod without any specific seed selected, then the spectating playfied doesn't match the one who is playing. I only have a very short clip, but it looks like this (look at the player at the bottom): 

https://user-images.githubusercontent.com/81645139/124313697-84290400-db71-11eb-8fb4-644d45129ed4.mp4

### PR DESCRIPTION
**Pull request by smoogipoo**
Fixes https://github.com/ppy/osu/issues/13751

I tried writing a test for this, but `TestSceneSpectator` currently works without instantiating the "playing" player which bypasses this whole `BeginPlaying` pathway. Making it work requires significant work so I'm excluding it for now.

### CODE REVIEW

**Comment by bdach** [comment]
Failing tests need attention here. The obvious solutions are to either ensure that all tests have `ScoreInfo` populated correctly (the beatmap in particular), or make `SpectatorClient` tolerant towards that by allowing null `ScoreInfo.Beatmap` and falling back to the provided `GameplayBeatmap` if so.

The former seems like a better solution given `Player` ensures that these properties are populated before `PrepareReplay()`, but I'm not 100% on that so I'll leave the decision up to you.

File: osu.Game.Tests/Visual/Gameplay/TestSceneReplayRecording.cs [hunk 1]
```diff
@@ -45,7 +47,11 @@ public TestSceneReplayRecording()
                     {
                         recordingManager = new TestRulesetInputManager(new TestSceneModSettings.TestRulesetInfo(), 0, SimultaneousBindingMode.Unique)
                         {
-                            Recorder = new TestReplayRecorder(new Score { Replay = replay })
+                            Recorder = new TestReplayRecorder(new Score
+                            {
+                                Replay = replay,
+                                ScoreInfo = { Beatmap = new TestBeatmap(new OsuRuleset().RulesetInfo).BeatmapInfo }
```
Comment (peppy) [inline osu.Game.Tests/Visual/Gameplay/TestSceneReplayRecording.cs:53]: Any reason for not using the already constructed beatmap? Seems to work okay (requires removing some using statements).

```suggestion
                                ScoreInfo = { Beatmap = gameplayBeatmap.BeatmapInfo }
```

File: osu.Game.Tests/Visual/Gameplay/TestSceneSpectatorPlayback.cs [hunk 2]
```diff
@@ -354,7 +356,7 @@ public enum TestAction
         internal class TestReplayRecorder : ReplayRecorder<TestAction>
         {
             public TestReplayRecorder()
-                : base(new Score())
+                : base(new Score { ScoreInfo = { Beatmap = new TestBeatmap(new OsuRuleset().RulesetInfo).BeatmapInfo } })
```
Comment (peppy) [inline osu.Game.Tests/Visual/Gameplay/TestSceneSpectatorPlayback.cs:359]:
```suggestion
                : base(new Score { ScoreInfo = { Beatmap = new BeatmapInfo() } })
```

Above can also use this if deemed more appropriate, just feels better as it's not actually meaningful for this test to make a full `TestBeatmap`.

File: osu.Game.Tests/Visual/Gameplay/TestSceneReplayRecorder.cs [hunk 3]
```diff
@@ -54,7 +56,11 @@ public class TestSceneReplayRecorder : OsuManualInputManagerTestScene
                     {
                         recordingManager = new TestRulesetInputManager(new TestSceneModSettings.TestRulesetInfo(), 0, SimultaneousBindingMode.Unique)
                         {
-                            Recorder = recorder = new TestReplayRecorder(new Score { Replay = replay })
+                            Recorder = recorder = new TestReplayRecorder(new Score
+                            {
+                                Replay = replay,
+                                ScoreInfo = { Beatmap = new TestBeatmap(new OsuRuleset().RulesetInfo).BeatmapInfo }
```
Comment (peppy) [inline osu.Game.Tests/Visual/Gameplay/TestSceneReplayRecorder.cs:62]:
```suggestion
                                ScoreInfo = { Beatmap = gameplayBeatmap.BeatmapInfo }
```

**Review by peppy — CHANGES_REQUESTED** [review]

**Review by peppy — APPROVED** [review]

**Pull request merged**